apps/api/src/tests/api.e2e.spec.ts

```diff
@@ -1,50 +1,86 @@
 import { AppModule } from '../app.module';
-import { CreateStoreDto } from '@meal-time/api-interfaces';
+import { CreateStoreDto, UpdateStoreDto } from '@meal-time/api-interfaces';
 import { INestApplication, ValidationPipe } from '@nestjs/common';
 import { Test } from '@nestjs/testing';
 import * as pactum from 'pactum';
 import { PrismaService } from '../services/prisma/prisma.service';
 
 describe('App e2e', () => {
-  let app: INestApplication;
-  let prisma: PrismaService;
-
-  beforeAll(async () => {
-    const moduleRef = await Test.createTestingModule({
-      imports: [AppModule],
-    }).compile();
-
-    app = moduleRef.createNestApplication();
-    app.useGlobalPipes(
-      new ValidationPipe({
-        whitelist: true,
-      })
-    );
-    await app.init();
-    await app.listen(3333);
-
-    prisma = app.get(PrismaService);
-    await prisma.cleanDb();
-    pactum.request.setBaseUrl('http://localhost:3333');
-  });
-
-  afterAll(() => {
-    app.close();
-  });
-
-  describe('Stores', () => {
-    describe('Create store', () => {
-      const dto: CreateStoreDto = {
-        name: 'Rema 1000',
-      };
-      it('should create store', async () => {
-        return pactum
-          .spec()
-          .post('/stores')
-          .withBody(dto)
-          .expectStatus(201)
-          .stores('storeId', 'id');
-      });
-    });
-  });
+	let app: INestApplication;
+	let prisma: PrismaService;
+
+	beforeAll(async () => {
+		const moduleRef = await Test.createTestingModule({
+			imports: [AppModule],
+		}).compile();
+
+		app = moduleRef.createNestApplication();
+		app.useGlobalPipes(
+			new ValidationPipe({
+				whitelist: true,
+			})
+		);
+		await app.init();
+		await app.listen(3333);
+
+		prisma = app.get(PrismaService);
+		await prisma.cleanDb();
+		pactum.request.setBaseUrl('http://localhost:3333');
+	});
+
+	afterAll(() => {
+		app.close();
+	});
+
+	describe('Stores', () => {
```

Comment:
All endpoints should also be tested for `edge cases` e.g. creating the same name twice resulting in an 403.

In general testing edge cases makes sure that our application remains fully functional and is secure. We always need to test the "happy path" as you have done here, but we need to ensure, that the application also works and returns relevant information back to the user, if something went wrong (this should always be tested as well)

Reply:
Want edgecases for the services unit tests?